indica/readers/salutils.py

```diff
@@ -35,23 +36,51 @@ class SALWarning(UserWarning):
 
 
 class SALUtils(BaseIO):
-    def __init__(self, pulse: int, server: str = "https://sal.jetdata.eu"):
-        self.NAMESPACE: Tuple[str, str] = ("jet", server)
+    def __init__(self, pulse: int, server: str = "https://sal.jet.uk"):
         self.pulse = pulse
         self._reader_cache_id = f"ppf:{server.replace('-', '_')}:{pulse}"
         self._client = SALClient(server)
         self._client.prompt_for_password = False
 
+    def requires_authentication(self) -> None:
+        return self._client.auth_required
+
+    def authenticate(self, name: Optional[str] = None) -> bool:
+        """Log onto the JET/SAL system to access data.
+
+        Parameters
+        ----------
+        name:
+            Your username when logging onto Heimdall.
+
+        Returns
+        -------
+        :
+            Indicates whether authentication was succesful.
+        """
+        if name is None:
+            name = getuser()
+        try:
+            self._client.authenticate(name, getpass())
+            return True
+        except AuthenticationFailed:
+            return False
+
+    def close(self) -> None:
+        """Ends connection to the SAL server from which PPF data is being
+        read."""
+        del self._client
+
     def _get_signal(
         self, uid: str, instrument: str, quantity: str, revision: RevisionLike
     ) -> Tuple[Signal, str]:
         """Gets the signal for the given INSTRUMENT (DDA in JET), at the
         given revision."""
         path = self.get_sal_path(
-            uid,
-            instrument,
-            quantity,
-            self.get_revision(uid, instrument, quantity, revision),
+            uid=uid,
+            instrument=instrument,
+            revision=self.get_revision(uid, instrument, revision),
```

Comment:
This is unnecessary, since the get_revision method is already called in the _read_database method of the datareader.